servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -901,6 +902,15 @@ private static Throwable unwrapThrowable(final Throwable t) {
             if (t instanceof DecoderException && (cause = t.getCause()) instanceof SSLException) {
                 return cause;
             }
+            if (t instanceof IOException && t.getMessage().contains("Connection reset by peer")) {
+                // TODO: this is messy and probably not the right place to intercept this.
+                // This is really a channel closed exception that happens when netty tries to read from a
+                // channel that is closed.
+                Throwable channelClosedException = StacklessClosedChannelException.newInstance(
+                        DefaultNettyConnection.class, "exceptionCaught(IOException)");
```

Comment:
My 2c would be that there is value in propagating the original exception. "Connection reset by peer" is quite popular exception for both Netty and standard Java I/O. People more or less understand what it means.

If we identified that it's a racy behavior, it's ok. Network events race. Consider improving the test assertion to accept any `IOException`. The goal of that test is only to verify that this use-case fails with an exception and doesn't hang or cause any other troubles.